### PR TITLE
Remove one layer of log softmax in discrete control

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/torch/distributions.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def log_prob(self, value):
         return torch.log(self.pdf(value) + EPSILON)
 
     def all_log_prob(self):
-        return torch.log(self.probs + EPSILON)
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/torch/distributions.py:118]: Would add a comment saying that self.logits are already normalized due to the action masking.

**Review by ervteng — APPROVED** [review]
LGTM, maybe add a comment

File: ml-agents/mlagents/trainers/torch/distributions.py (hunk 1 above)
Comment (vincentpierre) [reply ml-agents/mlagents/trainers/torch/distributions.py:118]: Why not modify the masking instead then ?

**Review by ervteng — APPROVED** [review]
